### PR TITLE
Transfer function widget

### PR DESCRIPTION
**Pull request by thewtex**
@jourdain please take a look

itk-vtk-image-viewer-transfer-function-snapshot

### CODE REVIEW

File: src/viewer.js [hunk 1]
```diff
@@ -121,12 +110,69 @@ function createViewer(container, data) {
...
+    handleColor: 'rgb(50, 150, 50)',
+    buttonDisableFillColor: 'rgba(255, 255, 255, 0.5)',
+    buttonDisableStrokeColor: 'rgba(0, 0, 0, 0.5)',
+    buttonStrokeColor: 'rgba(0, 0, 0, 1)',
+    buttonFillColor: 'rgba(255, 255, 255, 1)',
+    strokeWidth: 2,
+    activeStrokeWidth: 3,
+    buttonStrokeWidth: 1.5,
+    handleWidth: 3,
+    iconSize: 20, // Can be 0 if you want to remove buttons (dblClick for (+) / rightClick for (-))
```
Comment (jourdain) [inline src/viewer.js]: The buttons are kind of optional if you know (+) double click (-) right click.

File: src/viewer.js [hunk 2]
```diff
@@ -121,12 +110,69 @@ function createViewer(container, data) {
...
   const renderWindow = fullScreenRenderer.getRenderWindow();
 
+  const dataArray = data.image.getPointData().getScalars();
+  if (!dataArray) {
+    window.alert('No data array available in dataset');
+  }
+  const dataRange = dataArray.getRange();
+
+  const lookupTable = vtkColorTransferFunction.newInstance();
+  // TODO: Make colormap selectable
```
Comment (jourdain) [inline src/viewer.js]: Do you want me to try to improve the UI a little bit with draggable window and preset selection?

File: src/viewer.js (hunk 1 above)
Comment (thewtex) [reply src/viewer.js]: I think it is nice to have them explicitly there -- hard to guess that behavior otherwise.

Comment (jourdain) [reply src/viewer.js]: ok no worries.. Without them you can have slicker UI.

File: src/viewer.js (hunk 2 above)
Comment (thewtex) [reply src/viewer.js]: Yes, please!

Also, I was trying to figure out a way to limit the renderWindow size in pixels, for our container purposes, so it does not hammer the system and become very slow on high DPI screens. Any ideas?

Comment (jourdain) [reply src/viewer.js]: I'll give it a try tomorrow.

Are you saying the slowness depend on the resolution of the 3D window? or the piecewise widget?

Comment (thewtex) [reply src/viewer.js]: Thanks!

The 3D window -- fullscreen on a 4k display requests many pixels.

Comment (jourdain) [reply src/viewer.js]: I'll see if I can add some options to limit the resolution like we do in ParaViewWeb.

**Pull request merged**